go/enclave/rpc_server.go

```diff
@@ -151,16 +152,14 @@ func (s *server) SubmitTx(_ context.Context, request *generated.SubmitTxRequest)
 
 func (s *server) ExecuteOffChainTransaction(_ context.Context, request *generated.OffChainRequest) (*generated.OffChainResponse, error) {
 	result, err := s.enclave.ExecuteOffChainTransaction(request.EncryptedParams)
-	var errorJSON []byte
-	var merr error
 	if err != nil {
-		// TODO Review this error serialization
-		// serialise the error object returned by the evm into a json
-		errorJSON, merr = json.Marshal(err)
-		if merr != nil {
-			return nil, merr
+		// handle complex errors from the EVM
+		errResponse, processErr := serializeEVMError(err)
+		if processErr != nil {
+			// unable to serialize the error
+			return nil, fmt.Errorf("unable to handle complex error - %w", processErr)
```

Comment:
Can you at least provide some kind of string or type for the error you failed to serialise, to give the caller something to log?

Reply:
Not sure I follow, sorry. It's returning a new error with the reason of failing, I can improve that message, but I don't think that's what you're referring to ?

Reply:
I think you should return an error saying "tried to serialise the error. Failed. Original error message was blah"

Reply:
Yep 👍